com.unity.visualeffectgraph/Editor/Data/VFXData.cs

```diff
@@ -108,15 +107,17 @@ protected internal override void Invalidate(VFXModel model, InvalidationCause ca
                     owner.Invalidate(owner, cause);
         }
 
+        public override VFXGraph GetGraph()
+        {
+            if (m_Owners.Count > 0)
+                return m_Owners[0].GetGraph();
+            return null;
+        }
+
         public override void Sanitize(int version)
         {
             base.Sanitize(version);
-
-            if (m_Parent == null)
-            {
-                string assetPath = AssetDatabase.GetAssetPath(this);
-                m_Parent = VisualEffectResource.GetResourceAtPath(assetPath).GetOrCreateGraph();
-            }
+            m_Parent = null;
```

Comment:
Are we sure that having a null parent has no implication at all?

Reply:
To me, it means invalidation happening on the VFXData won't be propagated to parent graph anymore, which can cause issues

Reply:
The Invalidate is already iterating over owners and it doesn't need to send an invalidation to the main graph (see [here](https://github.com/Unity-Technologies/Graphics/blob/master/com.unity.visualeffectgraph/Editor/Data/VFXData.cs#L102). 
So far, I didn't find access to m_Parent in our current implementation
Let me double check this behavior (beside the implicit GetGraph implementation) ⏳

Reply:
I have double checked:

- m_Parent is acceded while calling AddChild/RemoveChild on VFXModel (see [here](https://github.com/Unity-Technologies/Graphics/blob/master/com.unity.visualeffectgraph/Editor/Models/VFXModel.cs#L157)), it will never be used for VFXData
- m_Parent is directly returned by [GetParent](https://github.com/Unity-Technologies/Graphics/blob/master/com.unity.visualeffectgraph/Editor/Models/VFXModel.cs#L196), beside editor test, it's used with [context](https://github.com/Unity-Technologies/Graphics/blob/fcffae0178a4cc07a9b600e35b2eb9723b69b631/com.unity.visualeffectgraph/Editor/GraphView/Elements/Controllers/VFXContextController.cs#L186)), in [VFXDataAnchorController](https://github.com/Unity-Technologies/Graphics/blob/fcffae0178a4cc07a9b600e35b2eb9723b69b631/com.unity.visualeffectgraph/Editor/GraphView/Elements/Controllers/VFXDataAnchorController.cs#L413) (we aren't expecting rendered slot on VFXData) and [VFXParameterNodeController](https://github.com/Unity-Technologies/Graphics/blob/fcffae0178a4cc07a9b600e35b2eb9723b69b631/com.unity.visualeffectgraph/Editor/GraphView/Elements/Controllers/VFXParameterNodeController.cs#L351) (which call it from a slot)

Then, we have a GetParent access in VFXModel with [GetFirstOfType](https://github.com/Unity-Technologies/Graphics/blob/fcffae0178a4cc07a9b600e35b2eb9723b69b631/com.unity.visualeffectgraph/Editor/Models/VFXModel.cs#L201) which is called either on owner of [VFXData](https://github.com/Unity-Technologies/Graphics/blob/f4079dcbbca4d0f847224140ac5644ff273306c1/com.unity.visualeffectgraph/Editor/Data/VFXData.cs#L285) or [VFXSlot](https://github.com/Unity-Technologies/Graphics/blob/e2cb5e6f60f2ef2047f0fd317ee512162c88360f/com.unity.visualeffectgraph/Editor/Compiler/VFXCodeGenerator.cs#L346)

All in all, the having a not declared parent is wrong approach, I think the main issue was about the GetGraph() behavior but it now returns the first graph of the first owner (see [here](https://github.com/Unity-Technologies/Graphics/pull/5976/files/a673f5c30fb34d307a22511ed836f571c980e4d2#diff-fe97c1cc58fb20b96676af540e5d17f8f1839b44cba93e99b2b11a8bdd180208R110))